--- BPMDisplay.cpp	Thu Jan 12 14:59:16 2006
+++ BPMDisplay.cp_	Sat Jul 29 15:33:30 2006
@@ -130,8 +130,8 @@
 
 	m_textBPM.StopTweening();
 	m_sprLabel->StopTweening();
-	m_textBPM.BeginTweening(0.5f);
-	m_sprLabel->BeginTweening(0.5f);
+	m_textBPM.BeginTweening(0.0f);
+	m_sprLabel->BeginTweening(0.0f);
 	if( GAMESTATE->IsExtraStage() || GAMESTATE->IsExtraStage2() )
 	{
 		m_textBPM.SetDiffuse( EXTRA_COLOR );
@@ -156,6 +156,9 @@
 	SetBPMRange( bpms );
 
 	m_fCycleTime = 0.2f;
+	m_textBPM.SetDiffuse( EXTRA_COLOR );
+	m_sprLabel->SetDiffuse( EXTRA_COLOR );
+
 }
 
 void BPMDisplay::NoBPM()
