--- FadingBanner.cp_	Thu Jan 12 13:59:16 2006
+++ FadingBanner.cpp	Sun Feb 24 22:33:47 2013
@@ -173,6 +173,24 @@
 	m_Banner[GetBackIndex()].LoadRandom();
 }
 
+void FadingBanner::LoadAbc()
+{
+	if( m_Banner[GetBackIndex()].GetTexturePath() == THEME->GetPathToG("Banner abc") )
+		return;
+
+	BeforeChange();
+	m_Banner[GetBackIndex()].LoadAbc();
+}
+
+void FadingBanner::LoadGroup()
+{
+	if( m_Banner[GetBackIndex()].GetTexturePath() == THEME->GetPathToG("Banner group") )
+		return;
+
+	BeforeChange();
+	m_Banner[GetBackIndex()].LoadGroup();
+}
+
 void FadingBanner::LoadFallback()
 {
 	BeforeChange();
