--- PrefsManager.h	Thu Jan 12 13:59:18 2006
+++ PrefsManager.h_	Mon Mar 21 05:22:28 2011
@@ -30,6 +30,7 @@
 	bool			m_bShowBanners;
 	enum BackgroundModes { BGMODE_OFF, BGMODE_ANIMATIONS, BGMODE_MOVIEVIS, BGMODE_RANDOMMOVIES } m_BackgroundMode;
 	int				m_iNumBackgrounds;
+	bool			m_bEnableBGAnim;
 	float			m_fBGBrightness;
 	bool			m_bHiddenSongs;
 	bool			m_bVsync;
@@ -137,6 +138,7 @@
 	bool			m_bEasterEggs;
 	int 			m_iMarvelousTiming;
 	bool			m_bEventMode;
+	bool			m_bArcadeEventMode;
 	int				m_iCoinsPerCredit;
 	int				m_iNumArcadeStages;
 
@@ -265,6 +267,7 @@
 	CString			m_sIgnoredMessageWindows;
 
 	CString			m_sCoursesToShowRanking;
+	float			m_fDebounceTime;
 
 	/* Debug: */
 	bool			m_bLogToDisk;
