--- ReceptorArrow.cp_	Sun Sep 05 18:48:54 2004
+++ ReceptorArrow.cpp	Mon Mar 04 20:10:58 2013
@@ -65,8 +65,9 @@
 
 void ReceptorArrow::DrawPrimitives()
 {
-	m_pReceptorGo->SetHidden( !GAMESTATE->m_bPastHereWeGo );
-	m_pReceptorWaiting->SetHidden( GAMESTATE->m_bPastHereWeGo );
+	bool insong = (GAMESTATE->m_fSongBeat >= 0);
+	m_pReceptorGo->SetHidden( !insong );
+	m_pReceptorWaiting->SetHidden( insong );
 	m_pPressBlock->SetHidden( !m_bIsPressed );
 	m_bIsPressed = false;	// it may get turned back on next update
 
