--- ScreenEvaluation.h_	Thu Jan 12 13:59:18 2006
+++ ScreenEvaluation.h	Thu Oct 11 21:40:34 2007
@@ -21,7 +21,7 @@
 #include "HighScore.h"
 #include "RageSound.h"
 
-const int MAX_SONGS_TO_SHOW = 5;	// In summary, we show last 3 stages, plus extra stages if passed
+const int MAX_SONGS_TO_SHOW = 7;	// In summary, we show last 3 stages, plus extra stages if passed
 enum JudgeLine { marvelous, perfect, great, good, boo, miss, ok, max_combo, error, NUM_JUDGE_LINES };
 enum StatsLine { jumps, holds, mines, hands, NUM_STATS_LINES };
 
@@ -84,6 +84,7 @@
 	// grade area
 	AutoActor			m_sprGradeFrame[NUM_PLAYERS];
 	GradeDisplay		m_Grades[NUM_PLAYERS];
+	GradeDisplay		m_Grades2[NUM_PLAYERS];
 	AutoActor			m_sprGrade[NUM_PLAYERS];
 
 	// graph area
@@ -114,6 +115,7 @@
 	// judgment area
 	Sprite				m_sprJudgeLabels[NUM_JUDGE_LINES];
 	BitmapText			m_textJudgeNumbers[NUM_JUDGE_LINES][NUM_PLAYERS];
+	BitmapText			m_textMaxCombo2[NUM_PLAYERS];
 
 	// stats area
 	AutoActor			m_sprStatsLabel[NUM_STATS_LINES];
