--- src/crypto51/iterhash.h	2013-03-11 00:15:49.000000000 +0100
+++ src/crypto51/iterhash.h	2013-03-14 08:11:16.000000000 +0100
@@ -92,7 +92,7 @@
 {
 	this->ThrowIfInvalidTruncatedSize(size);
 
-	PadLastBlock(this->BlockSize() - 2*sizeof(HashWordType));
+	this->PadLastBlock(this->BlockSize() - 2*sizeof(HashWordType));
 	CorrectEndianess(this->m_data, this->m_data, this->BlockSize() - 2*sizeof(HashWordType));
 
 	this->m_data[this->m_data.size()-2] = B::ToEnum() ? this->GetBitCountHi() : this->GetBitCountLo();
--- src/crypto51/algebra.cpp	2013-03-11 00:15:49.000000000 +0100
+++ src/crypto51/algebra.cpp	2013-03-14 08:30:47.000000000 +0100
@@ -54,7 +54,7 @@
 	Element g[3]={b, a};
 	unsigned int i0=0, i1=1, i2=2;
 
-	while (!Equal(g[i1], this->Identity()))
+	while (!this->Equal(g[i1], this->Identity()))
 	{
 		g[i2] = Mod(g[i0], g[i1]);
 		unsigned int t = i0; i0 = i1; i1 = i2; i2 = t;
