--- ArrowEffects.cpp	Thu Jan 12 14:59:16 2006
+++ ArrowEffects.cp_	Thu Jun 22 01:45:50 2006
@@ -132,7 +132,7 @@
 
 	if( fEffects[PlayerOptions::EFFECT_TIPSY] > 0 )
 		f += fEffects[PlayerOptions::EFFECT_TIPSY] * ( cosf( RageTimer::GetTimeSinceStart()*1.2f + iCol*1.8f) * ARROW_SIZE*0.4f );
-	return f;
+	return floor(f);
 }
 
 float ArrowGetYOffsetFromYPos( PlayerNumber pn, int iCol, float YPos, float fYReverseOffsetPixels )
