--- NoteField.cpp	Sun Jul 03 14:51:48 2005
+++ NoteField.cp_	Mon Aug 08 19:19:54 2005
@@ -579,8 +579,13 @@
 		NextDisplay = CurDisplay; ++NextDisplay;
 
 		// draw notes from furthest to closest
-		for( i=iLastIndexToDraw; i>=iFirstIndexToDraw; --i )	//	 for each row
+		for( int ii=(iLastIndexToDraw-iFirstIndexToDraw); ii>=0; --ii )	//	 for each row
 		{	
+			if (c != 2)
+				i = ii+iFirstIndexToDraw;
+			else
+				i = (iLastIndexToDraw-iFirstIndexToDraw)-ii+iFirstIndexToDraw;
+
 			TapNote tn = GetTapNote(c, i);
 			if( tn.type == TapNote::empty )	// no note here
 				continue;	// skip
