--- Player.h	Thu Jan 12 14:59:18 2006
+++ Player.h_	Sat Aug 05 19:42:57 2006
@@ -49,7 +49,7 @@
 	void ApplyWaitingTransforms();
 
 	static float GetMaxStepDistanceSeconds();
-
+	LifeMeter*		m_pLifeMeter;
 protected:
 	void UpdateTapNotesMissedOlderThan( float fMissIfOlderThanThisBeat );
 	void OnRowCompletelyJudged( int iStepIndex );
@@ -71,7 +71,7 @@
 	ArrowBackdrop	m_ArrowBackdrop;
 	NoteField*		m_pNoteField;
 
-	HoldJudgment	m_HoldJudgment[MAX_NOTE_TRACKS];
+	HoldJudgment	m_HoldJudgment[MAX_NOTE_TRACKS+1];
 
 	Judgment		m_Judgment;
 	ProTimingDisplay m_ProTimingDisplay;
@@ -81,7 +81,6 @@
 	AttackDisplay	m_AttackDisplay;
 
 	int m_iDCState;
-	LifeMeter*		m_pLifeMeter;
 	CombinedLifeMeter*		m_pCombinedLifeMeter;
 	ScoreDisplay*	m_pScoreDisplay;
 	ScoreDisplay*	m_pSecondaryScoreDisplay;
@@ -95,6 +94,8 @@
 	RageSound		m_soundMine;
 	RageSound		m_soundAttackLaunch;
 	RageSound		m_soundAttackEnding;
+
+	int			m_marvelousghost;
 };
 
 class Player : public PlayerMinus
