--- RageDisplay_D3D.cp_	Sat Sep 18 17:38:12 2004
+++ RageDisplay_D3D.cpp	Sun Mar 03 02:47:02 2013
@@ -922,8 +922,8 @@
 
 /* Use the default poly-based implementation.  D3D lines apparently don't support
  * AA with greater-than-one widths. */
-/*
-void RageDisplay_D3D::DrawLineStrip( const RageSpriteVertex v[], int iNumVerts, float LineWidth )
+
+void RageDisplay_D3D::DrawLineStripInternal( const RageSpriteVertex v[], int iNumVerts, float LineWidth )
 {
 	ASSERT( iNumVerts >= 2 );
 	g_pd3dDevice->SetRenderState( D3DRS_POINTSIZE, *((DWORD*)&LineWidth) );	// funky cast.  See D3DRENDERSTATETYPE doc
@@ -937,7 +937,7 @@
 	);
 	StatsAddVerts( iNumVerts );
 }
-*/
+
 
 void RageDisplay_D3D::ClearAllTextures()
 {
